code/__HELPERS/text.dm

```diff
@@ -463,7 +463,12 @@
 			text = replacetext(text, "\[signfont\]",		"<font face=\"[signfont]\"><i>")
 			text = replacetext(text, "\[/signfont\]",		"</i></font>")
 	if(sign)
-		text = replacetext(text, "\[sign\]",	"<font face=\"[signfont]\"><i>[user ? user.real_name : "Anonymous"]</i></font>")
+		if(istype(P, /obj/item/pen/chameleon)) // if we are using chameleon pen use fake name from the pen
+			var/obj/item/pen/chameleon/chameleon_pen = P
+			// small tip for a player if the left forge_name empty
+			text = replacetext(text, "\[sign\]",	"<font face=\"[signfont]\"><i>[chameleon_pen.forge_name ? chameleon_pen.forge_name : "No name was provided"]</i></font>")
```

Comment:
This needs admin logging of some sort so that Staff can tell if a signature is forged.

Reply:
If i understood you correctly, i have this done. So you technically should be able to check in VV who messed up with paper, should i redo this in some other way?
```
/obj/item/pen/chameleon/on_write(obj/item/paper/P, mob/user)
	P.forge_name_forger = user.name
```

Reply:
That's not something an admin should need to think to check, is the thing.
Logs mean that admins can make informed decisions on it instead of being blindsided if, say for example, someone's signature was on something that was rulebreaking, and they don't realize it was a forgery.

Reply:
It needs to be its own separate adminlog.

Reply:
Got it i will add some kind of normal logging.